src/ppm.cpp

```diff
@@ -130,52 +130,54 @@ PPM::~PPM()
 }
 
 void PPM::metadata_print (const string &topic, const RdKafka::Metadata *metadata) {
-
-    cout << "Metadata for " << (topic.empty() ? "" : "all topics")
-        << "(from broker "  << metadata->orig_broker_id()
-        << ":" << metadata->orig_broker_name() << endl;
+    ilogger->info("Metadata for {}(from broker {}: {}", (topic.empty() ? "" : "all topics"), metadata->orig_broker_id(), metadata->orig_broker_name());
 
     /* Iterate brokers */
-    cout << " " << metadata->brokers()->size() << " brokers:" << endl;
+    ilogger->info(" {} brokers:", metadata->brokers()->size());
 
     for ( auto ib : *(metadata->brokers()) ) {
-        cout << "  broker " << ib->id() << " at " << ib->host() << ":" << ib->port() << endl;
+        ilogger->info(" broker {} at {}: {}", ib->id(), ib->host(), ib->port());
     }
 
     /* Iterate topics */
-    cout << metadata->topics()->size() << " topics:" << endl;
+    ilogger->info("{} topics", metadata->topics()->size());
 
     for ( auto& it : *(metadata->topics()) ) {
-
-        cout << "  topic \""<< it->topic() << "\" with " << it->partitions()->size() << " partitions:";
+        ilogger->info(" topic \"{}\" with {} partitions:", it->topic(), it->partitions()->size());
 
         if (it->err() != RdKafka::ERR_NO_ERROR) {
-            cout << " " << err2str(it->err());
-            if (it->err() == RdKafka::ERR_LEADER_NOT_AVAILABLE) cout << " (try again)";
+            ilogger->error(" {}", err2str(it->err()));
```

Comment:
It looks like there's two loggers in here, "ilogger" and "elogger". I assume that's a stand-in for info logging and error logging. If that's the case should these error messages go to the elogger instead? I'm not sure the practicality of splitting it this way but it would allow a single location for error logs vs info level.

Reply:
That makes sense to me! I had assumed it meant "internal logger" and "external logger". Info and error seem more likely. In this case, the error messages should go to the elogger, yeah.

Reply:
The elogger is now utilized for error messages in this file.